packages/next/client/image.tsx

```diff
@@ -761,5 +762,7 @@ function defaultLoader({
     }
   }
 
-  return `${root}?url=${encodeURIComponent(src)}&w=${width}&q=${quality || 75}`
+  return `${normalizePathTrailingSlash(root)}?url=${encodeURIComponent(
+    src
+  )}&w=${width}&q=${quality || 75}`
 }
```

Comment:
Thanks! Please add an integration test so we don't regress in the future.

You can copy/paste this one and set `trailingSlash: true` 

https://github.com/vercel/next.js/blob/76dee14db47381e8daab8fc5be4f1314115af9e0/test/integration/image-optimizer/test/index.test.js#L1158-L1193